### PR TITLE
Adding Pull Req template

### PR DESCRIPTION
**Pull request by rotem2022**
Added file:
- pull_request_template.md

resolve #37

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Resolves #[relevant issue(s), e.g. 47]
+
+### Definition of Done
+*Feel free to copy this information from the connected issue.*
+
+#### Changelog
+
+- [ ] The CHANGELOG has been updated
+- [ ] This PR does not include user-facing changes and doesn't require a
+  CHANGELOG update
```
[the remark this answers is not in the document]
Comment (rotem2022) [reply .github/pull_request_template.md]: you are right, I updated the file so now it's suits better.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+### Implemented Changes
+
+*Describe how the desired outcome above has been achieved with this PR. In
+particular, consider:*
+
+- _What's changed? Why were these changes made?_
+- _How should the reviewer approach this PR_
+
+### Connected Issue/Story
```
Comment (ormergi) [inline .github/pull_request_template.md:14]: I would call it "Related Issue"

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+### Desired Outcome
```
Comment (ormergi) [inline .github/pull_request_template.md:1]: How about "Why you need this PR" or "What this PR does"?

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+*Please describe the desired outcome for this PR.  Said another way, what was
+the original request that resulted in these code changes?*
+
+### Implemented Changes
```
Comment (ormergi) [inline .github/pull_request_template.md:6]: I am not sure what this section should describe, how the change is implemented should be realized from the code diff.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+*Describe how the desired outcome above has been achieved with this PR. In
+particular, consider:*
+
+- _What's changed? Why were these changes made?_
+- _How should the reviewer approach this PR_
```
Comment (ormergi) [inline .github/pull_request_template.md:12]: I think this deserves its owns section (something like "Notes to reviewer")

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+### Connected Issue/Story
+
+Resolves #[relevant issue(s), e.g. 47]
+
+### Definition of Done
+
+*Feel free to copy this information from the connected issue.*
```
Comment (ormergi) [inline .github/pull_request_template.md:20]: I would pass omit this section, no need to copy and paste the issue information there is a section for referencing issues.

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+#### Test coverage
+
+- [ ] This PR includes new unit and integration tests to go with the code
+  changes
+- [ ] The changes in this PR do not require tests
+
```
Comment (ormergi) [inline .github/pull_request_template.md:27]: I would omit this section as well, tests (unit or e2e) should be an integral part of a PR that changes the code.
Also, this section would not make sense for PRs that change other things, e.g: docs, GitHub Actions manifests.

**Review by ormergi — CHANGES_REQUESTED** [review]
Hi Rotem, this is very nice I hope it will improve PRs quality.
Please see in-line comments.

**Comment by rotem2022** [comment]
> Hi Rotem, this is very nice I hope it will improve PRs quality. Please see in-line comments.

Hi Or, thank you for all the notes!

**Review by ygalblum — APPROVED** [review]
LGTM

**Review by ormergi — APPROVED** [review]
LGTM

**Pull request merged**